src/main/java/innohack/gem/service/extract/ExcelExtractor.java

```diff
@@ -12,6 +22,79 @@ public ExcelExtractor(ExtractConfig extractConfig) {
 
   @Override
   public ExtractedRecords extract(GEMFile f) throws Exception {
-    return null;
+    f.extract();
+
+    for (AbstractFeature feature : f.getData()) {
+
+      if (feature instanceof ExcelFeature) {
+        ExtractedRecords results = new ExtractedRecords();
+
+        List<String> excelSheetNames = getExtractConfig().getsheetNames();
+        List<String> extractColumns = getExtractConfig().getNamesColumn();
+        List<TimestampColumn> extractTSColumns = getExtractConfig().getTimestampColumn();
+
+        Map<String, List<List<String>>> sheetTables = ((ExcelFeature) feature).getSheetTableData();
+
+        for (String excelSheetName : excelSheetNames) {
+          if (sheetTables.containsKey(excelSheetName)) {
+            List<List<String>> rows = sheetTables.get(excelSheetName);
+
+            for (int i = 1; i < rows.size(); i++) results.getRecords().add(Lists.newArrayList());
+            // for getting the header for each sheet
+            List<String> columns = rows.get(0);
+
+            for (int j = 0; j < columns.size(); j++) {
+              String column = columns.get(j);
+              if (extractColumns.contains(column)) {
+                populate(results, rows, j, column);
+              }
+              for (TimestampColumn extractTSColumn : extractTSColumns) {
+                if (extractTSColumn.getFromColumn().equals(column)) {
+                  populate(
+                      results,
+                      rows,
+                      j,
+                      extractTSColumn.getName(),
+                      extractTSColumn.getFormat(),
+                      extractTSColumn.getTimezone());
+                }
+              }
+            }
+          }
+        }
+        return results;
+      }
+    }
+    throw new IllegalStateException("No Excel Feature found");
+  }
+
+  private void populate(
+      ExtractedRecords results, List<List<String>> rows, int columnIdx, String name)
+      throws ParseException {
+    populate(results, rows, columnIdx, name, null, null);
+  }
+
+  private void populate(
+      ExtractedRecords results,
+      List<List<String>> rows,
+      int columnIdx,
+      String name,
+      String dateFormat,
+      String timezone)
+      throws ParseException {
+    DateFormat formatter = null;
+    if (dateFormat != null) {
+      formatter = new SimpleDateFormat(dateFormat);
+      formatter.setTimeZone(TimeZone.getTimeZone(timezone));
+    }
+
+    results.getHeaders().add(name);
+    for (int i = 1; i < rows.size(); i++) {
+      String value = rows.get(i).get(columnIdx);
+      if (formatter != null) {
+        value = String.valueOf(formatter.parse(value).getTime());
```

Comment:
@orhlulu I have changed the timestamp column to give a predefined string format of the time, see `TimestampColumn.FORMAT`. This might make it easier for the UI visualization (no need to check whether need to format the timestamp value)

Reply:
ok so do i need to change anything from the excel extractor?

Reply:
Instead of giving the Timestamp in long value, I give it as `yyyy/MM/dd HH:mm:ssZ`, e.g. `2020/05/13 02:46:14+0800`

Reply:
ok.. can i make the changes here as well